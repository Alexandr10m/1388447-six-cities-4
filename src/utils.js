const firstWordInUpper = (str) => {
  const isString = typeof str === `string` || !!str;

  return isString ? str[0].toUpperCase() + str.slice(1) : str;
};

export {firstWordInUpper};
