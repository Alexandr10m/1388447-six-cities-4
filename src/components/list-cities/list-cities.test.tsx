import * as React from "react";
import * as renderer from "react-test-renderer";
import ListCities from "./list-cities";
import {BrowserRouter} from "react-router-dom";
import {noop} from "../../utils";


describe(`Snapshot of ListCities`, () => {
  it(`ListCities should render correctly`, () => {
    const tree = renderer
      .create(
          <BrowserRouter>
            <ListCities
              cities={[`Amsterdam`, `Dusseldorf`, `Hamburg`]}
              currentCity={`Amsterdam`}
              onCityClick={noop}
            />
          </BrowserRouter>)
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
