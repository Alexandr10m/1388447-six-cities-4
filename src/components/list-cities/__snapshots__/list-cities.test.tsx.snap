// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot of ListCities ListCities should render correctly 1`] = `
<ul
  className="locations__list tabs__list"
>
  <li
    className="locations__item"
    onClick={[Function]}
  >
    <a
      className="locations__item-link tabs__item tabs__item--active"
      href="/Amsterdam"
      onClick={[Function]}
    >
      <span>
        Amsterdam
      </span>
    </a>
  </li>
  <li
    className="locations__item"
    onClick={[Function]}
  >
    <a
      className="locations__item-link tabs__item "
      href="/Dusseldorf"
      onClick={[Function]}
    >
      <span>
        Dusseldorf
      </span>
    </a>
  </li>
  <li
    className="locations__item"
    onClick={[Function]}
  >
    <a
      className="locations__item-link tabs__item "
      href="/Hamburg"
      onClick={[Function]}
    >
      <span>
        Hamburg
      </span>
    </a>
  </li>
</ul>
`;
